payjoin/src/core/receive/v2/session.rs

```diff
@@ -96,42 +95,14 @@ impl SessionHistory {
     }
 
     /// Terminal error from the session if present
-    pub fn terminal_error(&self) -> Option<(String, Option<JsonReply>)> {
+    /// TODO: This should replay the event log and return the actual error, not a JSON reply
+    pub fn terminal_error(&self) -> Option<JsonReply> {
```

Comment:
Since this is returning a replyable json reply for now. Should we rename it to `json_reply`
?